### PR TITLE
Wolfenstein squash

### PR DESCRIPTION
**Pull request by endian-benjamin**

### CODE REVIEW

**Comment by linusnielsen** [comment]
The UART open-drain workaround needs to have a KConfig so it can be disabled on Catacomb.

File: drivers/modem/gsm_ppp.c [hunk 1]
```diff
@@ -412,22 +418,53 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
...
+		"AT+QNWINFO", &gsm->sem_response, K_SECONDS(2));
+#endif
+
+	/* Ensure PDP context is activated. This is likely a NOP for most
+	 * modem/SIM combos, but it is required for certain setups.
+	 */
+	(void)modem_cmd_send_nolock(
+		&gsm->context.iface, &gsm->context.cmd_handler,
+		NULL, 0, "AT+CGACT=1,1",
+		&gsm->sem_response, GSM_CMD_SETUP_TIMEOUT);
```
Comment (linusnielsen) [inline drivers/modem/gsm_ppp.c:466]: Can we do AT+CGACT=1,1 on other modems, or is this a Quectel only thing?

Comment (endian-benjamin) [reply drivers/modem/gsm_ppp.c:466]: It should be harmless but on SIMCOM the whole connection sequence breaks for some reason.

**Review by DunderRoffe — APPROVED** [review]

File: drivers/modem/gsm_ppp.c [hunk 2]
```diff
@@ -357,6 +357,12 @@ void __weak gsm_ppp_application_setup(struct modem_context *context,
 static void gsm_finalize_connection(struct gsm_modem *gsm)
 {
 	int ret;
+	static int attach_retry;
+
+	/* If attach check failed, we should not redo every setup step */
```
Comment (BassemMohsen) [inline drivers/modem/gsm_ppp.c:362]: We should test that for both Arkessa and Tele2 sims on 7600E to make sure it works fine.
Let me know if you need to verify it later.

**Pull request merged**